airflow/providers/databricks/hooks/databricks.py

```diff
@@ -168,13 +174,45 @@ def _do_api_call(self, endpoint_info, json):
 
         self.databricks_conn = self.get_connection(self.databricks_conn_id)
 
+        headers = USER_AGENT_HEADER.copy()
+
         if 'token' in self.databricks_conn.extra_dejson:
             self.log.info('Using token auth. ')
             auth = _TokenAuth(self.databricks_conn.extra_dejson['token'])
             if 'host' in self.databricks_conn.extra_dejson:
                 host = self._parse_host(self.databricks_conn.extra_dejson['host'])
             else:
                 host = self.databricks_conn.host
+        elif (
+            'azure_client_id' in self.databricks_conn.extra_dejson
+            and 'azure_client_secret' in self.databricks_conn.extra_dejson
+            and 'azure_tenant_id' in self.databricks_conn.extra_dejson
```

Comment:
Enforcing all three values to be present may result in unexpected behavior. Maybe we should raise an Exception if at least one, but not all three azure_* configs are set?

Reply:
good idea